app/scripts/ExportLinkDialog.js

```diff
@@ -1,26 +1,50 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+import Button from './Button';
 import Dialog from './Dialog';
 
 import '../styles/ExportLinkDialog.module.scss';
 
-const ExportLinkDialog = props => (
-  <Dialog
-    okayOnly={true}
-    okayTitle="Done"
-    onOkay={props.onDone}
-    title="Share view link"
-  >
-    <input
-      onClick={event => event.target.select()}
-      placeholder="Generating the link..."
-      readOnly={true}
-      styleName="full-width"
-      value={props.url}
-    />
-  </Dialog>
-);
+class ExportLinkDialog extends React.Component {
+  render() {
+    return (
+      <Dialog
+        okayOnly={true}
+        okayTitle="Done"
+        onOkay={this.props.onDone}
+        title="Share view link"
+      >
+        <p>
+          Copied to clipboard!
+          <span aria-label="Hooray" role="img"> 🎉</span>
+        </p>
+        <div styleName="export-link-dialog-wrapper">
+          <input
+            ref={(element) => {
+              this.input = element;
+              element.focus();
+              element.select();
+              document.execCommand('copy');
```

Comment:
Is there a precedent for a UI automatically placing something in the clipboard? I'm concerned that it might be annoying if unexpected.

Reply:
I was asking myself the same question but isn't the whole purpose of the link export modal to present you a URL that you copy?

Reply:
@nvictus what do you think? Since you started the call for more laziness in regards to the modal 😁

Reply:
How would anyone know of this auto-clipboarding side effect? I think a simple copy-to-clipboard button is something people are very accustomed to:

![image](https://user-images.githubusercontent.com/1270651/58662223-ae0aff00-82f7-11e9-9e00-ced3a66a800b.png)

Reply:
I can confirm, though, that the modal currently closes when clicking anywhere outside of it, which isn't typically how modals work. It makes it easy to accidentally lose the modal before getting a chance to copy the link.

That's probably the real source of frustration.